Make the nais-cli handle Fasit applicationProperties containing comments

The env operation of the nais-cli crashes if there is a Fasit alias for a resources of the type applicationProperties, and that property file contains one or more comments.

Crash output:
```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/nais/naisd/api.FasitClient.mapToNaisResource(0x10662b3, 0x15, 0xc42008c060, 0x7, 0xc42038a370, 0xa, 0x54cbb, 0xc4205dc040, 0x17, 0xc4205dc020, ...)
	/go/src/github.com/nais/naisd/api/fasit.go:557 +0x65f
github.com/nais/naisd/api.FasitClient.getScopedResource(0x10662b3, 0x15, 0xc42008c060, 0x7, 0xc42038a370, 0xa, 0xc420156380, 0x17, 0xc4201563e0, 0x15, ...)
	/go/src/github.com/nais/naisd/api/fasit.go:387 +0x43d
github.com/nais/naisd/api.FasitClient.GetScopedResources(0x10662b3, 0x15, 0xc42008c060, 0x7, 0xc42038a370, 0xa, 0xc42032aa00, 0x13, 0x20, 0x7ffe27700ecb, ...)
	/go/src/github.com/nais/naisd/api/fasit.go:162 +0x1d9
github.com/nais/naisd/api.(*FasitClient).GetScopedResources(0xc4203cac90, 0xc42032aa00, 0x13, 0x20, 0x7ffe27700ecb, 0x2, 0x7ffe27700ebb, 0xc, 0x7ffe27700eb3, 0x3, ...)
	<autogenerated>:1 +0xdd
github.com/nais/naisd/api.FetchFasitResources(0x170a7c0, 0xc4203cac90, 0x7ffe27700ebb, 0xc, 0x7ffe27700ecb, 0x2, 0x7ffe27700eb3, 0x3, 0xc42031d500, 0x12, ...)
	/go/src/github.com/nais/naisd/api/fasit.go:282 +0x3f8
github.com/nais/naisd/cli/cmd.glob..func2(0x1753880, 0xc420093770, 0x1, 0x5)
	/go/src/github.com/nais/naisd/cli/cmd/environment.go:128 +0xa92
github.com/nais/naisd/vendor/github.com/spf13/cobra.(*Command).execute(0x1753880, 0xc420093720, 0x5, 0x5, 0x1753880, 0xc420093720)
	/go/src/github.com/nais/naisd/vendor/github.com/spf13/cobra/command.go:757 +0x2c1
github.com/nais/naisd/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1753ae0, 0x18, 0xc42027df70, 0x406094)
	/go/src/github.com/nais/naisd/vendor/github.com/spf13/cobra/command.go:843 +0x334
github.com/nais/naisd/vendor/github.com/spf13/cobra.(*Command).Execute(0x1753ae0, 0xc4203ca150, 0x105c5a5)
	/go/src/github.com/nais/naisd/vendor/github.com/spf13/cobra/command.go:791 +0x2b
main.main()
	/go/src/github.com/nais/naisd/cli/nais.go:11 +0x31
```

If one tries to deploy the application through the nais deploy in jira one will only get a Bad gateway response:

```
2018-04-10 15:51:18,644 - REALLY_NAIS_DEPLOY - INFO - Calling deploy service...
2018-04-10 15:51:18,941 - REALLY_NAIS_DEPLOY - ERROR - HTTP Error 502: Bad Gateway
2018-04-10 15:51:18,943 - REALLY_NAIS_DEPLOY - ERROR - Bad Gateway
```

### Reproduce the issue
Nais-cli version: 234.0.0

The issue can be reproduced by using the nais.yaml file for saksoversikt:9.11, running the command `nais env` against the environments t and q. In the t environment the comment have been removed, but the comment is still present in the q environment.

#### Environment t - Working

This now works after the comment was removed from saksoversikt.properties in the t environment:
```
nais -z sbs env saksoversikt -e t5
Enter password for <user name>: 
...

```

#### Environment q - Does not work
Does not work, because the comment is still present i saksoversikt.properties in the q environment.

```
nais -z sbs env saksoversikt -e q5                                                              
Enter password for <user name>: 
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/nais/naisd/api.FasitClient.mapToNaisResource(0x10662b3, 0x15, 0xc42008c060, 0x7, 0xc42038a370, 0xa, 0x54cbb, 0xc4205dc040, 0x17, 0xc4205dc020, ...)
	/go/src/github.com/nais/naisd/api/fasit.go:557 +0x65f
github.com/nais/naisd/api.FasitClient.getScopedResource(0x10662b3, 0x15, 0xc42008c060, 0x7, 0xc42038a370, 0xa, 0xc420156380, 0x17, 0xc4201563e0, 0x15, ...)
	/go/src/github.com/nais/naisd/api/fasit.go:387 +0x43d
github.com/nais/naisd/api.FasitClient.GetScopedResources(0x10662b3, 0x15, 0xc42008c060, 0x7, 0xc42038a370, 0xa, 0xc42032aa00, 0x13, 0x20, 0x7ffe27700ecb, ...)
	/go/src/github.com/nais/naisd/api/fasit.go:162 +0x1d9
github.com/nais/naisd/api.(*FasitClient).GetScopedResources(0xc4203cac90, 0xc42032aa00, 0x13, 0x20, 0x7ffe27700ecb, 0x2, 0x7ffe27700ebb, 0xc, 0x7ffe27700eb3, 0x3, ...)
	<autogenerated>:1 +0xdd
github.com/nais/naisd/api.FetchFasitResources(0x170a7c0, 0xc4203cac90, 0x7ffe27700ebb, 0xc, 0x7ffe27700ecb, 0x2, 0x7ffe27700eb3, 0x3, 0xc42031d500, 0x12, ...)
	/go/src/github.com/nais/naisd/api/fasit.go:282 +0x3f8
github.com/nais/naisd/cli/cmd.glob..func2(0x1753880, 0xc420093770, 0x1, 0x5)
	/go/src/github.com/nais/naisd/cli/cmd/environment.go:128 +0xa92
github.com/nais/naisd/vendor/github.com/spf13/cobra.(*Command).execute(0x1753880, 0xc420093720, 0x5, 0x5, 0x1753880, 0xc420093720)
	/go/src/github.com/nais/naisd/vendor/github.com/spf13/cobra/command.go:757 +0x2c1
github.com/nais/naisd/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1753ae0, 0x18, 0xc42027df70, 0x406094)
	/go/src/github.com/nais/naisd/vendor/github.com/spf13/cobra/command.go:843 +0x334
github.com/nais/naisd/vendor/github.com/spf13/cobra.(*Command).Execute(0x1753ae0, 0xc4203ca150, 0x105c5a5)
	/go/src/github.com/nais/naisd/vendor/github.com/spf13/cobra/command.go:791 +0x2b
main.main()
	/go/src/github.com/nais/naisd/cli/nais.go:11 +0x31
```

--- a/api/fasit.go
+++ b/api/fasit.go
@@ -14,6 +14,7 @@ import (
 	"github.com/Jeffail/gabs"
 	"github.com/golang/glog"
 	"github.com/prometheus/client_golang/prometheus"
+	"regexp"
 )
 
 func init() {
@@ -548,10 +549,14 @@ func (fasit FasitClient) mapToNaisResource(fasitResource FasitResource, property
 		resource.certificates = files
 
 	} else if fasitResource.ResourceType == "applicationproperties" {
+		lineFilter, err := regexp.Compile(`[\p{L}\d_.]+=[\p{L}\d_,]+`)
+		if err != nil {
+			return NaisResource{}, fmt.Errorf("unable to compile regex: %s", err)
+		}
+
 		for _, line := range strings.Split(fasitResource.Properties["applicationProperties"], "\n") {
 			line = strings.TrimSpace(line)
-
-			if len(line) > 0 {
+			if lineFilter.MatchString(line) {
 				parts := strings.SplitN(line, "=", 2)
 				resource.properties[parts[0]] = parts[1]
 			}

--- a/api/fasit.go
+++ b/api/fasit.go
@@ -549,7 +549,7 @@ func (fasit FasitClient) mapToNaisResource(fasitResource FasitResource, property
 		resource.certificates = files
 
 	} else if fasitResource.ResourceType == "applicationproperties" {
-		lineFilter, err := regexp.Compile(`[\p{L}\d_.]+=[\p{L}\d_,]+`)
+		lineFilter, err := regexp.Compile(`^[\p{L}\d_.]+=.+`)
 		if err != nil {
 			return NaisResource{}, fmt.Errorf("unable to compile regex: %s", err)
 		}

--- a/api/fasit.go
+++ b/api/fasit.go
@@ -559,6 +559,8 @@ func (fasit FasitClient) mapToNaisResource(fasitResource FasitResource, property
 			if lineFilter.MatchString(line) {
 				parts := strings.SplitN(line, "=", 2)
 				resource.properties[parts[0]] = parts[1]
+			} else if (len(line) > 0) {
+				glog.Infof("the following string did not match our regex: %s", line)
 			}
 		}
 		delete(resource.properties, "applicationProperties")

--- a/api/fasit_test.go
+++ b/api/fasit_test.go
@@ -724,10 +724,11 @@ func TestGettingListOfResources(t *testing.T) {
 	assert.Equal(t, alias2, resourcesReplies[1].name)
 	assert.Equal(t, alias3, resourcesReplies[2].name)
 	assert.Equal(t, alias4, resourcesReplies[3].name)
-	assert.Equal(t, 3, len(resourcesReplies[3].properties))
+	assert.Len(t, resourcesReplies[3].properties, 4)
 	assert.Equal(t, "value1", resourcesReplies[3].properties["key1"])
 	assert.Equal(t, "dc=preprod,dc=local", resourcesReplies[3].properties["key2"])
-	assert.Equal(t, "value3", resourcesReplies[3].properties["key3"])
+	assert.Equal(t, "value3", resourcesReplies[3].properties["nøkkel3"])
+	assert.Equal(t, "værdi4", resourcesReplies[3].properties["key4"])
 }
 
 func TestResourceWithArbitraryPropertyKeys(t *testing.T) {

--- a/api/testdata/fasitResponse4.json
+++ b/api/testdata/fasitResponse4.json
@@ -7,7 +7,7 @@
     "application": "myapp"
   },
   "properties": {
-    "applicationProperties": "key1=value1\r\nkey2=dc=preprod,dc=local\nkey3=value3\n"
+    "applicationProperties": "key1=value1\r\nkey2=dc=preprod,dc=local\nnøkkel3=value3\n# en liten kommentar\nkey4=værdi4"
   },
   "secrets": {},
   "files": {},

